listener.go

```diff
@@ -28,7 +28,7 @@ func NewListener(l net.Listener) *GracefulListener {
 type gracefulConn struct {
 	net.Conn
 	lastHTTPState http.ConnState
-	forceClosed   bool
+	protected     bool
```

Comment:
This field is always set to `true` just before we call `s.StartRoutine()` and `false` just after we call `s.FinishRoutine()`. How about naming it `calledStart`?